engines/griffon/dialogs.cpp

```diff
@@ -224,27 +241,45 @@ void GriffonEngine::title(int mode) {
 	}
 }
 
+static Common::String formatPercent(int val) {
+	val = val / 2.55;
+	return Common::String::format("%d percent", val - val % 10);
+}
+
 void GriffonEngine::configMenu() {
-	static const char *optionTitles[22] = {
+	static const char *optionTitles[25] = {
 		"", "",
 		"", "", "", "",
 		"", "", "",
 		"Music:", "", "",
 		"Sound Effects:", "", "",
+		"Text to Speech:", "", "",
 		"Music Volume:", "",
 		"Effects Volume:", "", "", "", ""
 	};
-	static const char *optionValues[22] = {
+	static const char *optionValues[25] = {
 		"", "",
 		"", "", "", "",
 		"", "", "",
 		"On", "Off", "",
 		"On", "Off", "",
+		"On", "Off", "",
 		"[----------]", "",
 		"[----------]", "",
 		"Exit + Save", "",
 		"Exit"
 	};
+	static const int curselMapTitles[15] = {
```

Comment:
I am worried that this array might be too small and you may need to pad it with a couple of 0 to avoid out of bound access which could have unexpected consequences.

Looking at the code, it seems that both `curselMapTitles` and `curselMapValues` can be accessed with an index between `MINCURSEL` and `MAXCURSEL` inclusive. And `MAXCURSEL` is defined as 16. Since indexes start from 0, this means the arrays need to have at least 17 values to avoid an out of bound access when used with index 16.